src/nvim/option.c

```diff
@@ -3843,26 +3847,23 @@ set_bool_option (
   return NULL;
 }
 
-/*
- * Set the value of a number option, and take care of side effects.
- * Returns NULL for success, or an error message for an error.
- */
-static char_u *
-set_num_option (
-    int opt_idx,                            /* index in options[] table */
-    char_u *varp,                      /* pointer to the option variable */
-    long value,                             /* new value */
-    char_u *errbuf,                    /* buffer for error messages */
-    size_t errbuflen,                       /* length of "errbuf" */
-    int opt_flags                          /* OPT_LOCAL, OPT_GLOBAL and
-                                           OPT_MODELINE */
-)
+/// Set the value of a number option, and take care of side effects.
+/// Returns NULL for success, or an error message for an error.
+///
+/// @param opt_idx index in options[] table
+/// @param varp pointer to the option variable
+/// @param value new value
+/// @param effbuf buffer for error messages
+/// @param errbuflen length of "errbuf"
+/// @param opt_flags OPT_LOCAL, OPT_GLOBAL and OPT_MODELINE
+static char_u *set_num_option(int opt_idx, char_u *varp, NumOpt value,
```

Comment:
@cacplate I was thinking that you are in any case doing something with this function’s signature and since `varp` here is `char_u*` and function is `set_num_option` changing from `char_u*` to `void*` would not require much changes: you have to have casts in any case.